threads.go

```diff
@@ -541,11 +551,24 @@ func (t *threads) Subscribe(opts ...options.SubOption) tserv.Subscription {
 
 // putRecord adds an existing record. See PutOption for more.
 func (t *threads) putRecord(ctx context.Context, id thread.ID, lid peer.ID, rec thread.Record) error {
-	knownRecord, err := t.bstore.Has(rec.Cid())
-	if err != nil {
-		return err
+	unknownRecords := []thread.Record{}
+	cid := rec.Cid()
+	for cid.Defined() {
+		exist, err := t.bstore.Has(cid)
+		if err != nil {
+			return err
+		}
+		if exist {
+			break
+		}
+		r, err := t.GetRecord(ctx, id, cid)
+		if err != nil {
+			return err
+		}
+		unknownRecords = append(unknownRecords, r)
+		cid = r.PrevID()
 	}
-	if knownRecord {
+	if len(unknownRecords) == 0 {
```

Comment:
We had the received record? Nothing extra to do.

Further note: We're assuming that having a Record CID in the blockstore means we know that *the current log* has that record in it (and how knows, maybe not *other log*, since the blockstore is shared). That seems quite strange (two logs with a record with the same CID). Was this your original rationale too?

Reply:
Hmm, good point! At least currently where all events are encrypted with the single-use keys, it's not possible to duplicate a record across a log. If it was possible, I'm not sure how we'd be able to check if the record actually in _the log record chain_ w/o using a diff blockstore for each log. Maybe we'd have to use a new index in the datastore.

Reply:
I have a very cool gem to tell you about this blockstore assumption we did. Took me quite a long time to figure out that was a quite insane bug. 
I'll give further details about this in the next PR.